org.adempiere.base/src/org/compiere/model/MCostDetail.java

```diff
@@ -1624,6 +1624,14 @@ else if (getM_InOutLine_ID() != 0 		//	AR Shipment Detail Record
 			boolean addition = qty.signum() > 0;
 			boolean adjustment = getM_InventoryLine_ID() > 0 && qty.signum() == 0 && amt.signum() != 0;
 			boolean isVendorRMA = isVendorRMA();
+			
+			//If not import and it is due to inventory then don't mark as addition
+			if(addition && getM_InventoryLine_ID() != 0) {
+				int I_Inventory_ID = MInventoryLine.getImportLine_ID(getM_InventoryLine_ID(),get_TrxName());
+				if(I_Inventory_ID<=0)
+					addition = false;
```

Comment:
this is not always correct since inventory transaction can be either physical inventory or internal use inventory transaction.

Reply:
@dpansheriya , it is the other way round, I means it is possible that it is addition for internal use inventory.

Reply:
When record created using import inventory then it consider as addition in both the case of Physical invenotry or Internal use inentory. If stock is added using physical inventory or internal use inventory manually then it do not considered addition as it mostly some time missing stocks added back.

Reply:
@dpansheriya , that's true for physical inventory. For internal use, it is actual addition/subtraction of inventory, not for for missing stock.